source/client-side-encryption/client-side-encryption.rst

```diff
@@ -881,6 +921,44 @@ If the collection namespace has an associated ``encryptedFields``, then do the f
 - Create the collection ``collectionName`` with ``collectionOptions`` and the option ``encryptedFields`` set to the ``encryptedFields``.
 - Create the the index ``{"__safeContent__": 1}`` on collection ``collectionName``.
 
+
+Create Encrypted Collection Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To support automatic generation of encryption data keys, a helper
+`CreateEncryptedCollection` is defined, associated with a ClientEncryption_
+object `C`.
+
+For `CreateEncryptedCollection(C, collName, collOpts, kmsProvider, dkOpts)`,
+where `kmsProv` is a KMSProviderName_ and `dkOpts` is a DataKeyOpts_, with the
+name of the database as `dbName`, look up the encrypted fields `EF` for the
+collection as `GetEncryptedFields(collOpts, collName, dbName, false)`
+(`See here <GetEncryptedFields_>`_).
+
+If `EF` is *not-found*, report an error.
+
+Let `EF'` be a copy of `EF`. Update `EF'` in the following manner:
+
+- Let `Fields` be the ``"fields"`` element within `EF'`.
+- If `Fields` is present and is an array value, then for each element `F` of
+  `Fields`:
+
+  - If `F` is not a document element, skip it.
+  - Otherwise, if `F` has a ``"keyId"`` named element `K` and `K` is a ``null``
+    value:
+
+    - Let `D` be the result of ``C.createDataKey(kmsProv, dkOpts)``.
+    - If generating `D` resulted in an error `E`, the entire
+      `CreateEncryptedCollection` must now fail with error `E`.
```

Comment:
If an error occurs, it may be difficult to determine which keys were created before the error. A user may want to delete those keys.

Consider always returning `encryptedFields`. Even on error, created key ids can be included in the result.